server/app.ts

```diff
@@ -0,0 +1,63 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as express from "express";
+import * as compression from "compression";
+import * as bodyParser from "body-parser";
+
+import { graphqlExpress, graphiqlExpress } from 'graphql-server-express';
+import { makeExecutableSchema } from 'graphql-tools';
+
+import { PORT } from './common';
+import * as plugins from './plugins/api';
+import { GenericNotifier } from './plugins/api/GenericNotifier';
+import { APIReturn } from './plugins/api/APIReturn';
+
+const typeDefs = fs.readFileSync(path.resolve(__dirname, "../api.graphql"), "utf8");
+
+const app = express();
+
+app.use(compression());
+process.on("unhandledRejection", err => {
+	throw err;
+});
+
+interface IStatusReturn {
+	[medium: string]: [Promise<APIReturn>] | APIReturn; // String for generic unpacking error
```

Comment:
Should probs just be:
```typescript
interface IStatusReturn {
	[medium: string]: [APIReturn]; // String for generic unpacking error
}
```
Not sure if GraphQL unwraps nested promises, but  even if it does its more idiomatic to wait on all the promises first before taking their values and returning them, i.e. `Promise` should usually be the outermost type in your return type. So it would be: `send_message: async (prev: any, args: any): Promise<IStatusReturn>`